File: app/src/main/java/com/lomza/moviesroom/movie/MoviesListAdapter.java
package com.lomza.moviesroom.movie;

import android.content.Context;
import android.support.v4.app.DialogFragment;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.lomza.moviesroom.R;
import com.lomza.moviesroom.db.Director;
import com.lomza.moviesroom.db.Movie;
import com.lomza.moviesroom.db.MoviesDatabase;

import java.util.List;

import static com.lomza.moviesroom.movie.MovieSaveDialogFragment.TAG_DIALOG_MOVIE_SAVE;

/**
 * @author Antonina
 */

public class MoviesListAdapter extends RecyclerView.Adapter<MoviesListAdapter.MoviesViewHolder> {
    private LayoutInflater layoutInflater;
    private List<Movie> movieList;
    private Context context;

    public MoviesListAdapter(Context context) {
        this.layoutInflater = LayoutInflater.from(context);
        this.context = context;
    }

    public void setMovieList(List<Movie> movieList) {
        this.movieList = movieList;
        notifyDataSetChanged();
    }

    @Override
    public MoviesViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        final View itemView = layoutInflater.inflate(R.layout.item_list_movie, parent, false);
        return new MoviesViewHolder(itemView);
    }

    @Override
    public void onBindViewHolder(MoviesViewHolder holder, int position) {
        if (movieList == null) {
            return;
        }

        final Movie movie = movieList.get(position);
        if (movie != null) {
            holder.titleText.setText(movie.title);

            final Director director = MoviesDatabase.getDatabase(context).directorDao().findDirectorById(movie.directorId);
            final String directorFullName;
            if (director != null) {
                holder.directorText.setText(director.fullName);
                directorFullName = director.fullName;
            } else {
                directorFullName = "";
            }
            holder.itemView.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    DialogFragment dialogFragment = MovieSaveDialogFragment.newInstance(movie.title, directorFullName);
                    dialogFragment.show(((AppCompatActivity) context).getSupportFragmentManager(), TAG_DIALOG_MOVIE_SAVE);
                }
            });
        }
    }

    @Override
    public int getItemCount() {
        if (movieList == null) {
            return 0;
        } else {
            return movieList.size();
        }
    }

    static class MoviesViewHolder extends RecyclerView.ViewHolder {
        private TextView titleText;
        private TextView directorText;

        public MoviesViewHolder(View itemView) {
            super(itemView);

            titleText = itemView.findViewById(R.id.tvMovieTitle);
            directorText = itemView.findViewById(R.id.tvMovieDirectorFullName);
        }
    }
}

File: app/src/main/java/com/lomza/moviesroom/movie/MoviesListAdapter.kt
package com.lomza.moviesroom.movie

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.DialogFragment
import androidx.recyclerview.widget.RecyclerView
import com.lomza.moviesroom.R
import com.lomza.moviesroom.db.Movie
import com.lomza.moviesroom.db.MoviesDatabase
import com.lomza.moviesroom.movie.MovieSaveDialogFragment.Companion.newInstance
import com.lomza.moviesroom.movie.MoviesListAdapter.MoviesViewHolder

/**
 * @author Antonina
 */
class MoviesListAdapter(val context: Context) : RecyclerView.Adapter<MoviesViewHolder>() {

    private val layoutInflater: LayoutInflater = LayoutInflater.from(context)
    private var movieList: List<Movie>? = null

    fun setMovieList(movieList: List<Movie>?) {
        this.movieList = movieList
        notifyDataSetChanged()
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MoviesViewHolder {
        val itemView = layoutInflater.inflate(R.layout.item_list_movie, parent, false)
        return MoviesViewHolder(itemView)
    }

    override fun onBindViewHolder(holder: MoviesViewHolder, position: Int) {
        movieList?.let {
            val movie = it[position]
            holder.titleText.text = movie.title
            val director = MoviesDatabase.getDatabase(context).directorDao().findDirectorById(movie.directorId)
            val directorFullName: String
            if (director != null) {
                holder.directorText.text = director.fullName
                directorFullName = director.fullName
            } else {
                directorFullName = ""
            }
            holder.itemView.setOnClickListener { v: View? ->
                val dialogFragment: DialogFragment = newInstance(movie.title, directorFullName)
                dialogFragment.show(
                    (context as AppCompatActivity).supportFragmentManager,
                    MovieSaveDialogFragment.TAG_DIALOG_MOVIE_SAVE
                )
            }
        }
    }

    override fun getItemCount(): Int {
        return if (movieList == null) {
            0
        } else {
            movieList!!.size
        }
    }

    class MoviesViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val titleText: TextView = itemView.findViewById(R.id.tvMovieTitle)
        val directorText: TextView = itemView.findViewById(R.id.tvMovieDirectorFullName)
    }
}